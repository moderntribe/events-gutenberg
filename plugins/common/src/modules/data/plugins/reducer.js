/**
 * Internal dependencies
 */
import { types } from '@moderntribe/common/data/plugins';

export default ( state = {}, action ) => {
	switch ( action.type ) {
		case types.ADD_PLUGIN:
			return {
				...state,
				[ action.payload.name ]: true,
			};
		case types.REMOVE_PLUGIN:
			return {
				...state,
				[ action.payload.name ]: false,
			};
		default:
			return state;
	}
};
