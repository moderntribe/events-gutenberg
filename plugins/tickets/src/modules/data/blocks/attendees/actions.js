/**
 * Internal dependencies
 */
import * as types from './types';

export const setTitle = ( title ) => ( {
	type: types.SET_ATTENDEES_TITLE,
	payload: {
		title,
	},
} );

export const setInitialState = ( payload ) => ( {
	type: types.SET_ATTENDEES_INITIAL_STATE,
	payload,
} );
