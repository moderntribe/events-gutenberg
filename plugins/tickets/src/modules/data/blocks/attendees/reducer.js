/**
 * Internal dependencies
 */
import * as types from './types';

export const DEFAULT_STATE = {
	title: '',
};

export default ( state = DEFAULT_STATE, action ) => {
	switch ( action.type ) {
		case types.SET_ATTENDEES_TITLE:
			return {
				...state,
				title: action.payload.title,
			};
		default:
			return state;
	}
};
