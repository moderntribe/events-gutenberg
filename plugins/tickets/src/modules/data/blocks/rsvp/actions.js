/**
 * Internal dependencies
 */
import { types } from '@moderntribe/tickets/data/blocks/rsvp';

export const createRSVP = () => ( {
	type: types.CREATE_RSVP,
} );

export const initializeRSVP = () => ( {
	type: types.INITIALIZE_RSVP,
} );

export const deleteRSVP = () => ( {
	type: types.DELETE_RSVP,
} );

export const setRSVPId = ( id ) => ( {
	type: types.SET_RSVP_ID,
	payload: {
		id,
	},
} );

export const setRSVPSettingsOpen = ( settingsOpen ) => ( {
	type: types.SET_RSVP_SETTINGS_OPEN,
	payload: {
		settingsOpen,
	},
} );

export const setRSVPHasChanges = ( hasChanges ) => ( {
	type: types.SET_RSVP_HAS_CHANGES,
	payload: {
		hasChanges,
	},
} );

export const setRSVPIsLoading = ( isLoading ) => ( {
	type: types.SET_RSVP_IS_LOADING,
	payload: {
		isLoading,
	},
} );

export const setRSVPIsSettingsLoading = ( isSettingsLoading ) => ( {
	type: types.SET_RSVP_IS_SETTINGS_LOADING,
	payload: {
		isSettingsLoading,
	},
} );

export const setRSVPGoingCount = ( goingCount ) => ( {
	type: types.SET_RSVP_GOING_COUNT,
	payload: {
		goingCount,
	},
} );

export const setRSVPNotGoingCount = ( notGoingCount ) => ( {
	type: types.SET_RSVP_NOT_GOING_COUNT,
	payload: {
		notGoingCount,
	},
} );

export const setRSVPTitle = ( title ) => ( {
	type: types.SET_RSVP_TITLE,
	payload: {
		title,
	},
} );

export const setRSVPDescription = ( description ) => ( {
	type: types.SET_RSVP_DESCRIPTION,
	payload: {
		description,
	},
} );

export const setRSVPCapacity = ( capacity ) => ( {
	type: types.SET_RSVP_CAPACITY,
	payload: {
		capacity,
	},
} );

export const setRSVPNotGoingResponses = ( notGoingResponses ) => ( {
	type: types.SET_RSVP_NOT_GOING_RESPONSES,
	payload: {
		notGoingResponses,
	},
} );

export const setRSVPStartDate = ( startDate ) => ( {
	type: types.SET_RSVP_START_DATE,
	payload: {
		startDate,
	},
} );

export const setRSVPStartDateObj = ( startDateObj ) => ( {
	type: types.SET_RSVP_START_DATE_OBJ,
	payload: {
		startDateObj,
	},
} );

export const setRSVPEndDate = ( endDate ) => ( {
	type: types.SET_RSVP_END_DATE,
	payload: {
		endDate,
	},
} );

export const setRSVPEndDateObj = ( endDateObj ) => ( {
	type: types.SET_RSVP_END_DATE_OBJ,
	payload: {
		endDateObj,
	},
} );

export const setRSVPStartTime = ( startTime ) => ( {
	type: types.SET_RSVP_START_TIME,
	payload: {
		startTime,
	},
} );

export const setRSVPEndTime = ( endTime ) => ( {
	type: types.SET_RSVP_END_TIME,
	payload: {
		endTime,
	},
} );

export const setRSVPTempTitle = ( title ) => ( {
	type: types.SET_RSVP_TEMP_TITLE,
	payload: {
		title,
	},
} );

export const setRSVPTempDescription = ( description ) => ( {
	type: types.SET_RSVP_TEMP_DESCRIPTION,
	payload: {
		description,
	},
} );

export const setRSVPTempCapacity = ( capacity ) => ( {
	type: types.SET_RSVP_TEMP_CAPACITY,
	payload: {
		capacity,
	},
} );

export const setRSVPTempNotGoingResponses = ( notGoingResponses ) => ( {
	type: types.SET_RSVP_TEMP_NOT_GOING_RESPONSES,
	payload: {
		notGoingResponses,
	},
} );

export const setRSVPTempStartDate = ( startDate ) => ( {
	type: types.SET_RSVP_TEMP_START_DATE,
	payload: {
		startDate,
	},
} );

export const setRSVPTempStartDateObj = ( startDateObj ) => ( {
	type: types.SET_RSVP_TEMP_START_DATE_OBJ,
	payload: {
		startDateObj,
	},
} );

export const setRSVPTempEndDate = ( endDate ) => ( {
	type: types.SET_RSVP_TEMP_END_DATE,
	payload: {
		endDate,
	},
} );

export const setRSVPTempEndDateObj = ( endDateObj ) => ( {
	type: types.SET_RSVP_TEMP_END_DATE_OBJ,
	payload: {
		endDateObj,
	},
} );

export const setRSVPTempStartTime = ( startTime ) => ( {
	type: types.SET_RSVP_TEMP_START_TIME,
	payload: {
		startTime,
	},
} );

export const setRSVPTempEndTime = ( endTime ) => ( {
	type: types.SET_RSVP_TEMP_END_TIME,
	payload: {
		endTime,
	},
} );

export const setRSVPHeaderImage = ( payload ) => ( {
	type: types.SET_RSVP_HEADER_IMAGE,
	payload,
} );

export const setRSVPDetails = ( payload ) => ( {
	type: types.SET_RSVP_DETAILS,
	payload,
} );

export const setRSVPTempDetails = ( payload ) => ( {
	type: types.SET_RSVP_TEMP_DETAILS,
	payload,
} );
