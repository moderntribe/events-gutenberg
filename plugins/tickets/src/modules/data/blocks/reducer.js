/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import rsvp from './rsvp';
import ticket from './ticket';

export default combineReducers( {
	rsvp,
	ticket,
} );
