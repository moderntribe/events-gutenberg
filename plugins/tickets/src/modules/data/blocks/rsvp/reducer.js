/**
 * External dependencies
 */
import moment from 'moment/moment';

/**
 * Internal dependencies
 */
import { types } from '@moderntribe/tickets/data/blocks/rsvp';
import { moment as momentUtil } from '@moderntribe/common/utils';

export const DEFAULT_STATE = {
	title: '',
	description: '',
	capacity: '',
	notGoingResponses: false,
	startDate: momentUtil.toDate( moment() ),
	endDate: momentUtil.toDate( moment() ),
	startTime: momentUtil.toTime24Hr( moment() ),
	endTime: momentUtil.toTime24Hr( moment() ),
	tempTitle: '',
	tempDescription: '',
	tempCapacity: '',
	tempNotGoingResponses: false,
	tempStartDate: momentUtil.toDate( moment() ),
	tempStartDateObj: new Date( momentUtil.toDate( moment() ) ),
	tempEndDate: momentUtil.toDate( moment() ),
	tempEndDateObj: new Date( momentUtil.toDate( moment() ) ),
	tempStartTime: momentUtil.toTime24Hr( moment() ),
	tempEndTime: momentUtil.toTime24Hr( moment() ),
	created: false,
	settingsOpen: false,
	hasChanges: false,
};

export default ( state = DEFAULT_STATE, action ) => {
	switch ( action.type ) {
		case types.CREATE_RSVP:
			return {
				...state,
				created: true,
			};
		case types.DELETE_RSVP:
			return DEFAULT_STATE;
		case types.SET_RSVP_TITLE:
			return {
				...state,
				title: action.payload.title,
			};
		case types.SET_RSVP_DESCRIPTION:
			return {
				...state,
				description: action.payload.description,
			};
		case types.SET_RSVP_CAPACITY:
			return {
				...state,
				capacity: action.payload.capacity,
			};
		case types.SET_RSVP_NOT_GOING_RESPONSES:
			return {
				...state,
				notGoingResponses: action.payload.notGoingResponses,
			};
		case types.SET_RSVP_START_DATE:
			return {
				...state,
				startDate: action.payload.startDate,
			};
		case types.SET_RSVP_END_DATE:
			return {
				...state,
				endDate: action.payload.endDate,
			};
		case types.SET_RSVP_START_TIME:
			return {
				...state,
				startTime: action.payload.startTime,
			};
		case types.SET_RSVP_END_TIME:
			return {
				...state,
				endTime: action.payload.endTime,
			};
		case types.SET_RSVP_TEMP_TITLE:
			return {
				...state,
				tempTitle: action.payload.tempTitle,
			};
		case types.SET_RSVP_TEMP_DESCRIPTION:
			return {
				...state,
				tempDescription: action.payload.tempDescription,
			};
		case types.SET_RSVP_TEMP_CAPACITY:
			return {
				...state,
				tempCapacity: action.payload.tempCapacity,
			};
		case types.SET_RSVP_TEMP_NOT_GOING_RESPONSES:
			return {
				...state,
				tempNotGoingResponses: action.payload.tempNotGoingResponses,
			};
		case types.SET_RSVP_TEMP_START_DATE:
			return {
				...state,
				tempStartDate: action.payload.tempStartDate,
			};
		case types.SET_RSVP_TEMP_START_DATE_OBJ:
			return {
				...state,
				tempStartDateObj: action.payload.tempStartDateObj,
			};
		case types.SET_RSVP_TEMP_END_DATE:
			return {
				...state,
				tempEndDate: action.payload.tempEndDate,
			};
		case types.SET_RSVP_TEMP_END_DATE_OBJ:
			return {
				...state,
				tempEndDateObj: action.payload.tempEndDateObj,
			};
		case types.SET_RSVP_TEMP_START_TIME:
			return {
				...state,
				tempStartTime: action.payload.tempStartTime,
			};
		case types.SET_RSVP_TEMP_END_TIME:
			return {
				...state,
				tempEndTime: action.payload.tempEndTime,
			};
		case types.SET_RSVP_SETTINGS_OPEN:
			return {
				...state,
				settingsOpen: action.payload.settingsOpen,
			};
		case types.SET_RSVP_HAS_CHANGES:
			return {
				...state,
				hasChanges: action.payload.hasChanges,
			};
		default:
			return state;
	}
};
