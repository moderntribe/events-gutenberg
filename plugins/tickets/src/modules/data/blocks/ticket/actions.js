/**
 * Internal dependencies
 */
import { types } from '@moderntribe/tickets/data/blocks/ticket';

export const setHeader = ( header ) => ( {
	type: types.SET_TICKET_HEADER,
	payload: {
		header,
	},
} );

export const setTotalSharedCapacity = ( sharedCapacity ) => ( {
	type: types.SET_TICKET_TOTAL_SHARED_CAPACITY,
	payload: {
		sharedCapacity,
	},
} );

export const setSettingsOpen = ( isSettingsOpen ) => ( {
	type: types.SET_TICKET_SETTINGS_OPEN,
	payload: {
		isSettingsOpen,
	},
} );

export const openSettings = () => setSettingsOpen( true );
export const closeSettings = () => setSettingsOpen( false );

export const setParentBlockSelected = ( selected ) => ( {
	type: types.SET_PARENT_BLOCK_SELECTED,
	payload: {
		selected,
	},
} );

export const setChildBlockSelected = ( selected ) => ( {
	type: types.SET_CHILD_BLOCK_SELECTED,
	payload: {
		selected,
	},
} );

export const setActiveChildBlockId = ( activeChildBlockId ) => ( {
	type: types.SET_ACTIVE_CHILD_BLOCK_ID,
	payload: {
		activeChildBlockId,
	},
} );

export const setTempSharedCapacity = ( sharedCapacity ) => ( {
	type: types.SET_TICKET_TMP_TICKET_SHARED_CAPACITY,
	payload: {
		sharedCapacity,
	},
} );

export const setInitialState = ( props ) => ( {
	type: types.SET_INITIAL_STATE,
	payload: props,
} );

export const setProvider = ( provider ) => ( {
	type: types.SET_PROVIDER,
	payload: {
		provider,
	},
} );

// individual ticket actions
export const registerTicketBlock = ( blockId ) => ( {
	type: types.SET_TICKET_BLOCK_ID,
	payload: {
		blockId,
	},
} );

export const removeTicketBlock = ( blockId ) => ( {
	type: types.REMOVE_TICKET_BLOCK,
	payload: {
		blockId,
	},
} );

export const setTitle = ( blockId, title ) => ( {
	type: types.SET_TICKET_TITLE,
	payload: {
		blockId,
		title,
	},
} );

export const setDescription = ( blockId, description ) => ( {
	type: types.SET_TICKET_DESCRIPTION,
	payload: {
		blockId,
		description,
	},
} );

export const setPrice = ( blockId, price ) => ( {
	type: types.SET_TICKET_PRICE,
	payload: {
		blockId,
		price,
	},
} );

export const setSKU = ( blockId, SKU ) => ( {
	type: types.SET_TICKET_SKU,
	payload: {
		blockId,
		SKU,
	},
} );

export const setStartDate = ( blockId, startDate ) => ( {
	type: types.SET_TICKET_START_DATE,
	payload: {
		blockId,
		startDate,
	},
} );

export const setStartTime = ( blockId, startTime ) => ( {
	type: types.SET_TICKET_START_TIME,
	payload: {
		blockId,
		startTime,
	},
} );

export const setEndDate = ( blockId, endDate ) => ( {
	type: types.SET_TICKET_END_DATE,
	payload: {
		blockId,
		endDate,
	},
} );

export const setEndTime = ( blockId, endTime ) => ( {
	type: types.SET_TICKET_END_TIME,
	payload: {
		blockId,
		endTime,
	},
} );

export const setCapacityType = ( blockId, capacityType ) => ( {
	type: types.SET_TICKET_CAPACITY_TYPE,
	payload: {
		blockId,
		capacityType,
	},
} );

export const setCapacity = ( blockId, capacity ) => ( {
	type: types.SET_TICKET_CAPACITY,
	payload: {
		blockId,
		capacity,
	},
} );

export const createNewTicket = ( blockId ) => ( {
	type: types.SET_CREATE_NEW_TICKET,
	payload: {
		blockId,
	},
} );

export const updateTicket = ( blockId ) => ( {
	type: types.SET_UPDATE_TICKET,
	payload: {
		blockId,
	},
} );

export const setTicketIsEditing = ( blockId, isEditing ) => ( {
	type: types.SET_TICKET_IS_EDITING,
	payload: {
		blockId,
		isEditing,
	},
} );

export const setTicketId = ( blockId, postId ) => ( {
	type: types.SET_TICKET_ID,
	payload: {
		blockId,
		postId,
	},
} );

export const setTicketDateIsPristine = ( blockId, dateIsPristine ) => ( {
	type: types.SET_TICKET_DATE_PRISTINE,
	payload: {
		blockId,
		dateIsPristine,
	},
} );

export const setParentBlockIsLoading = ( isParentBlockLoading ) => ( {
	type: types.SET_PARENT_BLOCK_LOADING,
	payload: {
		isParentBlockLoading,
	},
} );

export const setTicketStartDateMoment = ( blockId, startDateMoment ) => ( {
	type: types.SET_TICKET_START_DATE_MOMENT,
	payload: {
		blockId,
		startDateMoment,
	},
} );

export const setTicketEndDateMoment = ( blockId, endDateMoment ) => ( {
	type: types.SET_TICKET_END_DATE_MOMENT,
	payload: {
		blockId,
		endDateMoment,
	},
} );
