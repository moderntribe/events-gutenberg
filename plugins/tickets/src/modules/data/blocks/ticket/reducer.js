/**
 * Internal dependencies
 */
import { types } from '@moderntribe/tickets/data/blocks/ticket';

export const DEFAULT_STATE = {
	sharedCapacity: '',
	header: null,
	isSettingsOpen: false,
};

export default ( state = DEFAULT_STATE, action ) => {
	switch ( action.type ) {
		case types.SET_TICKET_HEADER:
			return {
				...state,
				header: action.payload.header,
			};
		case types.SET_TICKET_SHARED_CAPACITY:
			return {
				...state,
				sharedCapacity: action.payload.sharedCapacity,
			};
		case types.SET_TICKET_SETTINGS_OPEN:
			return {
				...state,
				isSettingsOpen: action.payload.isSettingsOpen,
			};
		default:
			return state;
	}
}
