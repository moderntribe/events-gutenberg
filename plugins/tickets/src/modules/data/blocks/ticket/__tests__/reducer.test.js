/**
 * Internal dependencies
 */
import reducer, { DEFAULT_STATE } from '../reducer';
import * as actions from '../actions';

describe( 'Reducer', () => {
	it( 'should set the default state', () => {
		expect( reducer( undefined, {} ) ).toEqual( DEFAULT_STATE );
	} );

	it( 'should set the header', () => {
		expect( reducer(
			DEFAULT_STATE,
			actions.setTicketsHeader( { header: 1 } ),
		) ).toMatchSnapshot();
	} );

	it( 'should set the is settings open', () => {
		expect( reducer(
			DEFAULT_STATE,
			actions.setTicketsIsSettingsOpen( true ),
		) ).toMatchSnapshot();
	} );

	it( 'should set the is parent block loading', () => {
		expect( reducer(
			DEFAULT_STATE,
			actions.setTicketsIsParentBlockLoading( true ),
		) ).toMatchSnapshot();
	} );

	it( 'should set the is child block selected', () => {
		expect( reducer(
			DEFAULT_STATE,
			actions.setTicketsIsChildBlockSelected( true ),
		) ).toMatchSnapshot();
	} );

	it( 'should set the is parent block selected', () => {
		expect( reducer(
			DEFAULT_STATE,
			actions.setTicketsIsParentBlockSelected( true ),
		) ).toMatchSnapshot();
	} );

	it( 'should set the active child block id', () => {
		expect( reducer(
			DEFAULT_STATE,
			actions.setTicketsActiveChildBlockId( 'modern-tribe' ),
		) ).toMatchSnapshot();
	} );

	it( 'should set the provider', () => {
		expect( reducer(
			DEFAULT_STATE,
			actions.setTicketsProvider( 'provider' ),
		) ).toMatchSnapshot();
	} );

	it( 'should set the shared capacity', () => {
		expect( reducer(
			DEFAULT_STATE,
			actions.setTicketsSharedCapacity( 99 ),
		) ).toMatchSnapshot();
	} );

	it( 'should set the temp shared capacity', () => {
		expect( reducer(
			DEFAULT_STATE,
			actions.setTicketsTempSharedCapacity( 99 ),
		) ).toMatchSnapshot();
	} );
} );
