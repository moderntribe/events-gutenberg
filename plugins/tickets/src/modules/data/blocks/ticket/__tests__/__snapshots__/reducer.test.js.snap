// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reducer should set the active child block id 1`] = `
Object {
  "activeChildBlockId": "modern-tribe",
  "header": null,
  "isChildBlockSelected": false,
  "isParentBlockLoading": false,
  "isParentBlockSelected": false,
  "isSettingsOpen": false,
  "provider": "",
  "sharedCapacity": "",
  "tempSharedCapacity": "",
  "tickets": Object {
    "allIds": Array [],
    "byId": Object {},
  },
}
`;

exports[`Reducer should set the header 1`] = `
Object {
  "activeChildBlockId": "",
  "header": 1,
  "isChildBlockSelected": false,
  "isParentBlockLoading": false,
  "isParentBlockSelected": false,
  "isSettingsOpen": false,
  "provider": "",
  "sharedCapacity": "",
  "tempSharedCapacity": "",
  "tickets": Object {
    "allIds": Array [],
    "byId": Object {},
  },
}
`;

exports[`Reducer should set the is child block selected 1`] = `
Object {
  "activeChildBlockId": "",
  "header": null,
  "isChildBlockSelected": true,
  "isParentBlockLoading": false,
  "isParentBlockSelected": false,
  "isSettingsOpen": false,
  "provider": "",
  "sharedCapacity": "",
  "tempSharedCapacity": "",
  "tickets": Object {
    "allIds": Array [],
    "byId": Object {},
  },
}
`;

exports[`Reducer should set the is parent block loading 1`] = `
Object {
  "activeChildBlockId": "",
  "header": null,
  "isChildBlockSelected": false,
  "isParentBlockLoading": false,
  "isParentBlockSelected": undefined,
  "isSettingsOpen": false,
  "provider": "",
  "sharedCapacity": "",
  "tempSharedCapacity": "",
  "tickets": Object {
    "allIds": Array [],
    "byId": Object {},
  },
}
`;

exports[`Reducer should set the is parent block selected 1`] = `
Object {
  "activeChildBlockId": "",
  "header": null,
  "isChildBlockSelected": false,
  "isParentBlockLoading": undefined,
  "isParentBlockSelected": false,
  "isSettingsOpen": false,
  "provider": "",
  "sharedCapacity": "",
  "tempSharedCapacity": "",
  "tickets": Object {
    "allIds": Array [],
    "byId": Object {},
  },
}
`;

exports[`Reducer should set the is settings open 1`] = `
Object {
  "activeChildBlockId": "",
  "header": null,
  "isChildBlockSelected": false,
  "isParentBlockLoading": false,
  "isParentBlockSelected": false,
  "isSettingsOpen": true,
  "provider": "",
  "sharedCapacity": "",
  "tempSharedCapacity": "",
  "tickets": Object {
    "allIds": Array [],
    "byId": Object {},
  },
}
`;

exports[`Reducer should set the provider 1`] = `
Object {
  "activeChildBlockId": "",
  "header": null,
  "isChildBlockSelected": false,
  "isParentBlockLoading": false,
  "isParentBlockSelected": false,
  "isSettingsOpen": false,
  "provider": "provider",
  "sharedCapacity": "",
  "tempSharedCapacity": "",
  "tickets": Object {
    "allIds": Array [],
    "byId": Object {},
  },
}
`;

exports[`Reducer should set the shared capacity 1`] = `
Object {
  "activeChildBlockId": "",
  "header": null,
  "isChildBlockSelected": false,
  "isParentBlockLoading": false,
  "isParentBlockSelected": false,
  "isSettingsOpen": false,
  "provider": "",
  "sharedCapacity": 99,
  "tempSharedCapacity": "",
  "tickets": Object {
    "allIds": Array [],
    "byId": Object {},
  },
}
`;

exports[`Reducer should set the temp shared capacity 1`] = `
Object {
  "activeChildBlockId": "",
  "header": null,
  "isChildBlockSelected": false,
  "isParentBlockLoading": false,
  "isParentBlockSelected": false,
  "isSettingsOpen": false,
  "provider": "",
  "sharedCapacity": "",
  "tempSharedCapacity": 99,
  "tickets": Object {
    "allIds": Array [],
    "byId": Object {},
  },
}
`;
